packages/flutter/lib/src/material/stepper.dart

```diff
@@ -938,3 +999,168 @@ class _TrianglePainter extends CustomPainter {
     );
   }
 }
+
+/// The `StepStyle` class provides customization options for configuring the appearance of a step in a stepper widget.
+///
+/// To create a customized step style, instantiate a `StepStyle` object and pass it to the `Step` widget.
+///
+/// Example usage:
+/// ```dart
+/// Step(
+///   title: const Text('Step 1'),
+///   content: const Text('Content for Step 1'),
+///   stepStyle: StepStyle(
+///     color: Colors.blue,
+///     errorColor: Colors.red,
+///     border: Border.all(color: Colors.grey),
+///     boxShadow: const BoxShadow(blurRadius: 3.0, color: Colors.black26),
+///     gradient: const LinearGradient(colors: <Color>[Colors.red, Colors.blue]),
+///     indexStyle: const TextStyle(color: Colors.white),
+///   ),
+/// )
+/// ```
+/// See also:
+/// {@tool dartpad}
+/// An example the shows how to use the [StepStyle], and the [StepStyle] UI
```

Comment:
An example that uses [StepStyle] to customize the appearance of each [Step] in a [Stepper].